apis/projectcontour/v1/httpproxy.go

```diff
@@ -40,6 +44,11 @@ type HTTPProxySpec struct {
 	// is given precedence over this field.
 	// +optional
 	IngressClassName string `json:"ingressClassName,omitempty"`
+
+	// HttpVersions specify the http versions to offer for this HTTPProxy.
+	// If empty, the DefaultHTTPVersions from v1alpha1.EnvoyConfig will be used.
+	// It is ignored when TCPProxy is set.
+	HttpVersions []HttpVersion `json:"httpVersions,omitempty"`
```

Comment:
I think this should move at least under the `VirtualHost` struct, since it is only applicable for root HTTPProxies that define a virtualhost. Still thinking about whether it should move under `VirtualHost.TLS` since it is really only relevant for TLS vhosts AFAIK. cc @sunjayBhatia and @tsaarni for any thoughts here.

Reply:
+1 for moving this under VirtualHost